src/funds_recovery_kit.rs

```diff
@@ -0,0 +1,1147 @@
+use std::fmt;
+
+use bitcoin::bip32;
+use bitcoin::consensus::encode;
+use bitcoin::hashes::hex::FromHex;
+use bitcoin::hashes::{sha512, Hash, HashEngine, Hmac, HmacEngine};
+use bitcoin::secp256k1::{Scalar, Secp256k1, SecretKey};
+use bitcoin::sighash;
+use bitcoin::sighash::EcdsaSighashType;
+use bitcoin::PrivateKey;
+use bitcoin::PublicKey;
+use bitcoin::Witness;
+use bitcoin::{Amount, Script, Transaction};
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+struct Event {
+    commitment_tx: String,
+    sweep_tx: String,
+    htlc_tx: Vec<String>,
+    commitment_number: u64,
+    channel_point: String,
+    sweep_tx_add_tweak: String,
+    htlc_tx_add_tweak: String,
+    serialized_htlc_sweep_tx: Vec<String>,
+    funding_private_key_derivation_path: String,
+    delayed_payment_base_key_derivation_path: String,
+    htlc_base_key_derivation_path: String,
+    channel_capacity: u64,
+    nonces: Vec<String>,
+    counterparty_sweep_tx: Option<String>,
+    counterparty_htlc_tx_add_tweak: Option<String>,
+    counterparty_serialized_htlc_sweep_tx: Option<Vec<String>>,
+    counterparty_nonces: Option<Vec<String>>,
+    payment_key_derivation_path: Option<String>,
+    counterparty_serialized_commitment_tx: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct Response {
+    pub commitment_tx: String,
+    pub sweep_tx: String,
+    pub htlc_inbound_tx: Vec<(String, String)>,
+    pub htlc_outbound_tx: Vec<(String, String)>,
```

Comment:
Tuples doesn't work with uniffi. I think you'll need to define a new type contains two strings here.